### PR TITLE
Revert back to platform-dependent tar decompression

### PR DESCRIPTION
**Pull request by timaa2k**
## High-level description

This PR reverts the decompression part of: [#4388](https://github.com/dcos/dcos/pull/4388/files)
After the aforementioned PR was merged we noticed that the CI runs into this Python bug:
https://bugs.python.org/issue21872

Due to complications to reproduce the issue locally this should be reverted to the old functionality for now.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS-48220](https://jira.mesosphere.com/browse/DCOS-48220) pkgpanda: Compressed file ended before the end-of-stream marker was reached.

## Checklist for all PRs

  - [ ] Added a comprehensible changelog entry to `CHANGES.md` or explain why this is not a user-facing change:
  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

### CODE REVIEW

**Comment by timaa2k** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by adamtheturtle** [comment]
I can confirm that all unit tests pass on macOS! Thank you @timaa2k !

File: pkgpanda/util.py [hunk 1]
```diff
@@ -218,8 +219,13 @@ def extract_tarball(path, target):
         assert os.path.exists(path), "Path doesn't exist but should: {}".format(path)
         make_directory(target)
 
-        with tarfile.open(name=str(path), mode='r') as tar:
-            tar.extractall(path=str(target), numeric_owner=True)
```
Comment (adamtheturtle) [inline pkgpanda/util.py]: Do we lose anything by no longer specifying numeric owner anywhere?

Comment (timaa2k) [reply pkgpanda/util.py]: We might, I guess only CI can answer that question at the moment.

Comment (adamtheturtle) [reply pkgpanda/util.py]: And if we find some edge case later we find missing tests - we must be willing to take those risks else we will be paralyzed by the past.

Comment (timaa2k) [reply pkgpanda/util.py]: The numeric owner option is applied when creating the archive in our case via the `tarfile.TarInfo`.
It is in the archive and the decompression is the same as before so I guess we do not lose anything.

Comment (adamtheturtle) [reply pkgpanda/util.py]: Is it? Before we touched anything there was the UID and GID set but there was also an option which forced numeric owner:

```
--numeric-owner
always use numbers for user/group names
```

Comment (timaa2k) [reply pkgpanda/util.py]: I've checked it by looking into a tar'd DC/OS package (using `dcos-shell` tar) before and after the change.
Before the change there is no non-numeric owner and group information stored:
```
$ tar -tvJf bootstrap--8704a6496c6dfe56980c4926f40e94b31da7f067.tar.xz
-rwxr-xr-x 0/0            2568 2018-03-22 19:45 ./bin/bootstrap
```
Afterwards there is also only numeric information about the owner:
```
$ tar -tvJf bootstrap--8704a6496c6dfe56980c4926f40e94b31da7f067.tar.xz
-rwxr-xr-x 0/0            2568 2019-01-28 17:27 ./bin/bootstrap
```
On extraction if the superuser extracts a tar archive the information in the archive will be used.
Look for `--same-owner` in the manual:
https://www.gnu.org/software/tar/manual/html_node/Attributes.html#SEC138
For non-root users this applies (from the same manual):
```
When restoring, [tar] tries to look the name (if one was written) up in `/etc/passwd'. If it fails, then it uses the user ID stored in the archive instead.
```

Comment (timaa2k) [reply pkgpanda/util.py]: `bsdtar` has the same semantics if running as `root`:
https://www.freebsd.org/cgi/man.cgi?query=bsdtar&sektion=1&manpath=freebsd-release-ports

**Comment by timaa2k** [comment]
@mesosphere-mergebot sync

**Comment by timaa2k** [comment]
@mesosphere-mergebot label Ship It

**Comment by timaa2k** [comment]
@mesosphere-mergebot merge-it
